UI/cypress/support/commands.js

```diff
@@ -24,23 +24,23 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
 
-import { removeNewLine } from "./utils";
-
-import { getJumpToLocationId } from "./utils";
+import {
+  removeNewLine,
+  getJumpToLocationId,
+  applyBasicAuth,
+  applyBasicAuthWithRequest,
+} from "./utils";
 
 Cypress.Commands.overwrite("visit", (originalFn, url, options) => {
-  const username = Cypress.env("username");
-  const password = Cypress.env("password");
+  options = applyBasicAuth(options);
+  return originalFn(url, options);
+});
 
-  if (username && password) {
-    options = options || {};
-    options.auth = {
-      username: username,
-      password: password,
-    };
+Cypress.Commands.overwrite("request", (originalFn, options) => {
+  if (Cypress.config().baseUrl.includes("find-a-tuition-partner")) {
```

Comment:
This seems wrong - the baseUrl is always going to be the same regardless of whether the url we're requesting is our site or not isn't it? What we need to be checking is whether the URL we're requesting (which may or may not be an external site) is our site, and only sending the credentials if it is ours. Otherwise we risk sending our credentials to random other sites.

Reply:
This is what I was trying to do with my bit of code here: https://github.com/DFE-Digital/find-a-tuition-partner/pull/327#discussion_r1179633489

Reply:
Okay, I have now incorporated your [suggested changes](https://github.com/DFE-Digital/find-a-tuition-partner/pull/327#discussion_r1179633489) to this PR. Made a slight tweak on the below:
```
if(typeof args[0] == 'object') to
if (typeof args == "object") 
```
Cypress will enforce basic authentication on all requests to the base URL after overwriting the "request" command with these modifications.